src/lib.rs

```diff
@@ -1048,4 +1065,64 @@ mod tests {
             r#"SendError { recipient_name: "TestActor", reason: Disconnected }"#
         );
     }
+
+    #[test]
+    fn message_priorities() {
+        env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("trace")).init();
+
+        struct PriorityActor {
+            received: Arc<Mutex<Vec<usize>>>,
+        }
+
+        impl Actor for PriorityActor {
+            type Context = Context<Self::Message>;
+            type Error = ();
+            type Message = usize;
+
+            fn handle(
+                &mut self,
+                context: &mut Self::Context,
+                message: Self::Message,
+            ) -> Result<(), Self::Error> {
+                let mut received = self.received.lock();
+                received.push(message);
+                if received.len() >= 20 {
+                    context.system_handle.shutdown().unwrap();
+                }
+                Ok(())
+            }
+
+            fn name() -> &'static str {
+                "PriorityActor"
+            }
+
+            fn priority(message: &Self::Message) -> Priority {
+                if *message >= 10 {
+                    Priority::High
+                } else {
+                    Priority::Normal
+                }
+            }
+        }
+
+        let addr = Addr::with_capacity(10);
+        let received = Arc::new(Mutex::new(Vec::<usize>::new()));
+
+        // Send messages before even actor starts.
+        for message in 0..20usize {
+            addr.send(message).unwrap();
+        }
+
+        let mut system = System::new("priorities");
+        system
+            .prepare(PriorityActor { received: Arc::clone(&received) })
+            .with_addr(addr)
+            .run_and_block()
+            .unwrap();
+
+        assert_eq!(
+            *received.lock(),
+            [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+        );
```

Comment:
This nicely fails when one enables `eventual-fairness` on flume:
```
---- tests::message_priorities stdout ----
thread 'tests::message_priorities' panicked at 'assertion failed: `(left == right)`
  left: `[0, 10, 11, 12, 1, 2, 13, 14, 15, 16, 17, 3, 18, 4, 5, 19, 6, 7, 8, 9]`,
 right: `[10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]`', src/lib.rs:1123:9
```

Reply:
Nice :) Thank you for confirming!